File: examples/oidc/public/rest-api/nodejs/cors.js
/**
 * Example CORS middleware. Do not use this configuration in production,
 * the origin is intentionally lax to work with different SPA examples.
 */
const CORS = (req, res, next) => {
  res.header('Access-Control-Allow-Origin', '*');
  res.header('Access-Control-Allow-Methods', 'GET, POST, PUT, DELETE, OPTIONS');
  res.header('Access-Control-Allow-Headers', 'Authorization');
  res.header('Access-Control-Allow-Credentials', 'true');
  if (req.method === 'OPTIONS') {
    return res.status(200).end();
  }
  next();
};

module.exports = {
    CORS,
}

File: examples/oidc/public/rest-api/golang/cors.go
package main

import (
	"net/http"
)

// CORS Middleware to handle preflight requests and allow Authorization header.
// Ensure to tighten up this policy for deployments, the origin is intentionally
// lax to work with the different SPA examples locally.
func CORS(next http.Handler) http.Handler {
	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
		w.Header().Set("Access-Control-Allow-Origin", "*")
		w.Header().Set("Access-Control-Allow-Methods", "GET,OPTIONS")
		w.Header().Set("Access-Control-Allow-Headers", "Authorization")
		w.Header().Set("Access-Control-Allow-Credentials", "true")

		if r.Method == http.MethodOptions {
			return
		}

		next.ServeHTTP(w, r)
	})
}
